zenodo/modules/spam/models.py

```diff
@@ -59,20 +59,27 @@ class SafelistEntry(db.Model):
 
     user = db.relationship(User, backref='safelist')
 
+    @classmethod
+    def get_by_user_id(cls, user_id):
+        """Get entry by user_id."""
+        try:
+            entry = cls.query.filter(cls.user_id == user_id).first()
+        except Exception:
+            entry = None
+
+        return entry
+
     @classmethod
     def create(cls, user_id, notes=None):
         """Create a safelist entry."""
         try:
-            entry = cls(user_id=user_id, notes=notes)
-            db.session.add(entry)
-            return entry
+            existing_entry = cls.get_by_user_id(user_id=user_id)
+            if not existing_entry:
+                entry = cls(user_id=user_id, notes=notes)
+                db.session.add(entry)
```

Comment:
minor: Instead of baking this check in the `.create`, I would check in the view `if user.safelist:`, to avoid checking/fetching twice.